 1) Allows to use node names without hostname. If node name doesn't
    contain @ then local hostname is added to the end of node name.
    If erlang is running with -name option then FDQN is added, else
    only hostname part is added. This patch closes bug #389635

 2) ejabberdctl script in Debian distribution has slightly different
    syntax than in upstream. It allows to omit ejabberd node name
    assuming ejabberd@hostname. The patch changes usage messages to
    reflect these changes in the syntax.

Index: ejabberd-1.1.4/src/ejabberd_ctl.erl
===================================================================
--- ejabberd-1.1.4.orig/src/ejabberd_ctl.erl	2006-05-08 01:26:06.000000000 +0400
+++ ejabberd-1.1.4/src/ejabberd_ctl.erl	2007-09-05 13:48:06.000000000 +0400
@@ -24,7 +24,21 @@
 start() ->
     case init:get_plain_arguments() of
 	[SNode | Args] ->
-	    Node = list_to_atom(SNode),
+	    case string:tokens(SNode, "@") of
+		[_Node, _Server] ->
+		    SNode1 = SNode;
+		_ ->
+		    SNode1 = case net_kernel:longnames() of
+				 true ->
+				     SNode ++ "@" ++ inet_db:gethostname() ++
+					      "." ++ inet_db:res_option(domain);
+				 false ->
+				     SNode ++ "@" ++ inet_db:gethostname();
+				 _ ->
+				     SNode
+			     end
+	    end,
+	    Node = list_to_atom(SNode1),
 	    Status = case rpc:call(Node, ?MODULE, process, [Args]) of
 			 {badrpc, Reason} ->
 			     io:format("RPC failed on the node ~p: ~p~n",
@@ -248,13 +262,14 @@
 		   Desc, NewLine]
 	  end, CmdDescs),
     io:format(
-      "Usage: ejabberdctl node command~n"
+      "Usage: ejabberdctl [--node node] command [options]~n"
       "~n"
       "Available commands:~n"
       ++ FmtCmdDescs ++
       "~n"
-      "Example:~n"
-      "  ejabberdctl ejabberd@host restart~n"
+      "Examples:~n"
+      "  ejabberdctl restart~n"
+      "  ejabberdctl --node ejabberd@host restart~n"
      ).
 
 print_vhost_usage(Host) ->
@@ -279,7 +294,7 @@
 		   Desc, NewLine]
 	  end, CmdDescs),
     io:format(
-      "Usage: ejabberdctl node vhost host command~n"
+      "Usage: ejabberdctl [--node node] vhost host command~n"
       "~n"
       "Available commands:~n"
       ++ FmtCmdDescs ++
